zebra-chain/src/work/difficulty.rs

```diff
@@ -74,15 +73,20 @@ impl fmt::Debug for CompactDifficulty {
 /// The precise bit pattern of an `ExpandedDifficulty` value is
 /// consensus-critical, because it is compared with the `block::Hash`.
 ///
-/// Note that each `CompactDifficulty` value represents a range of
-/// `ExpandedDifficulty` values, because the precision of the
-/// floating-point format requires rounding on conversion.
+/// Note that each `CompactDifficulty` value can be converted from a
+/// range of `ExpandedDifficulty` values, because the precision of
+/// the floating-point format requires rounding on conversion.
 ///
 /// Therefore, consensus-critical code must perform the specified
 /// conversions to `CompactDifficulty`, even if the original
 /// `ExpandedDifficulty` values are known.
+///
+/// Callers should avoid constructing `ExpandedDifficulty` zero
+/// values, because they are rejected by the consensus rules.
+//
+// TODO: Use NonZeroU256, when available
 #[derive(Clone, Copy, Eq, PartialEq, Ord, PartialOrd)]
-pub struct ExpandedDifficulty(U256);
+pub struct ExpandedDifficulty(pub(super) U256);
```

Comment:
this `pub(super)` makes me think that we might want to make a new `arbitrary` submodule for just this file rather than exposing our inner value to part of the crate graph

Reply:
I'm not quite sure how this would look, can you fill in some more of the details?
I find Rust modules confusing sometimes.

Reply:
items defined in a module are also visible to all sub modules regardless of the visibility modifier so you wouldn't need pub super if the code that accesses these methods was moved to a sub module of this module

Reply:
Fixed in 06ce32931848a090fea1a7072c823736c62c5027, including CompactDifficulty, ExpandedDifficulty, and Work.